src/test/java/net/starschema/clouddb/jdbc/BQScrollableResultSetFunctionTest.java

```diff
@@ -722,7 +727,48 @@ public void TestResultSetRelative() {
     public void TestResultSetTotalBytesProcessedCacheHit() {
         Assert.assertTrue(Result instanceof BQScrollableResultSet);
         BQScrollableResultSet results = (BQScrollableResultSet)Result;
-        Assert.assertEquals(results.getTotalBytesProcessed() == 0, results.getCacheHit());
+        final Boolean processedNoBytes = new Long(0L).equals(results.getTotalBytesProcessed());
+        Assert.assertEquals(processedNoBytes, results.getCacheHit());
+    }
+
+    @Test
+    public void testHandlesAllNullResponseFields() throws Exception {
+        try {
+            mockResponse("{}");
+        } catch (BQSQLException e) {
+            Assert.assertTrue(e.getMessage().contains("without a job reference"));
+            return;
+        }
+        throw new AssertionError("Expected graceful failure due to lack of job reference");
     }
 
+    @Test
+    public void testHandlesSomeNullResponseFields() throws Exception {
+        // Make sure we don't get any NPE's due to null values;
+        mockResponse(
+            "{ \"jobComplete\": true, "
+            + "\"totalRows\": \"0\", "
+            + "\"rows\": [] }");
+    }
+
+    private void mockResponse(String jsonResponse) throws Exception {
+        Properties properties =
+            BQSupportFuncts.readFromPropFile(
+                getClass().getResource("/installedaccount.properties").getFile());
+        String url = BQSupportFuncts.constructUrlFromPropertiesFile(properties, true, null);
+        // Mock an empty response object.
+        MockHttpTransport mockTransport =
+            new MockHttpTransport.Builder()
+                .setLowLevelHttpResponse(
+                    new MockLowLevelHttpResponse().setContent(jsonResponse))
+                .build();
+        BQConnection bq = new BQConnection(url + "&useLegacySql=false", properties, mockTransport);
+        BQStatement stmt = new BQStatement(properties.getProperty("projectid"), bq, ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
+        String sqlStmt = "SELECT word from publicdata:samples.shakespeare LIMIT 100";
+
+        BQScrollableResultSet results = ((BQScrollableResultSet)stmt.executeQuery(sqlStmt));
+
+        results.getTotalBytesProcessed();
+        results.getCacheHit();
```

Comment:
So we're just verify that there aren't any NPEs then?

Reply:
Oh nvrmind, missed the opening commend :D

Reply:
Ya, I'm suspicious that defaulting to `false` and 0 for `cache_hit` and `total_bytes_processed`, respectively, is creating some reporting issues. This could help explain why the cache hit rate we're calculating from the BQ job metadata logs (~40%) is so different from the hit rate we're calculating from these pinger metrics (~16%); defaulting to `false` when it should really be "no comment".

Reply:
Ok. When we integrate this new build of the bq driver, will we be able to test this from HT?

Reply:
^ happy to help in the ht-side testing effort. Good call.